### PR TITLE
buku: 3.8 -> 4.1

### PR DESCRIPTION
**Pull request by ck3d**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/buku/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
 { stdenv, python3, fetchFromGitHub, fetchpatch }:
```
Comment (dotlambda) [inline pkgs/applications/misc/buku/default.nix]: To circumvent the dateparser problem:
```suggestion
{ stdenv, python36, fetchFromGitHub, fetchpatch }:
```
and add a comment with the reason.
This should be done in a separate commit referencing https://github.com/NixOS/nixpkgs/issues/52766 in the commit message so that it can easily be reverted.

Comment (ck3d) [reply pkgs/applications/misc/buku/default.nix]: Good idea. I tried it, but now I got an odd problem. The depending modules are still using python37!
```
❯ nix build -f . buku                                                                                     ~/nixpkgs
builder for '/nix/store/c30nlyv9wjwzizd7x37hvh4cmqn4k8ma-python3.7-dateparser-0.7.0.drv' failed with exit code 1; last 10 log lines:
  - M
  + Miércoles 31 Diciembre 2014


  ----------------------------------------------------------------------
  Ran 23161 tests in 101.451s

  FAILED (failures=3, skipped=7)
  Test failed: <unittest.runner.TextTestResult run=23161 errors=0 failures=3>
  error: Test failed: <unittest.runner.TextTestResult run=23161 errors=0 failures=3>
cannot build derivation '/nix/store/dpvjy2zrn6px51j9kg64vrpnpi32ljgq-python3.7-maya-0.3.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/d1pmqfwddlw0627vgi2hli14168d6x5g-python3.7-Flask-Common-0.3.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/r6w3vdf98wbwd4b67x1cqbbj62d0c1lq-python3.7-httpbin-0.6.2.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/c24lh992njn2534iv57ykkmknii7wayf-python3.7-pytest-httpbin-0.3.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/p6lhmcprvvh6smy4hy3cnk7vr46lmlkw-python3.7-vcrpy-2.0.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/5jq53l30nic9ifsvx8y6l4giynhxk6x7-buku-4.1.drv': 1 dependencies couldn't be built
[0 built (1 failed)]
error: build of '/nix/store/5jq53l30nic9ifsvx8y6l4giynhxk6x7-buku-4.1.drv' failed
```
I have no clue how this can happen.

Comment (dotlambda) [reply pkgs/applications/misc/buku/default.nix]: That's not possible. It must have been a mistake on your side. Everything's working fine on my side.

**Comment by dotlambda** [comment]
Now we still have

> still not working

in the commit message :(

File: pkgs/applications/misc/buku/default.nix (hunk 1 above)
Comment (ck3d) [reply pkgs/applications/misc/buku/default.nix]: You are right, I built buku from my overlay and not the patched one.
I have to keep in mind to deactivate all overlays!
